docs/manual/called-methods-checker.tex

```diff
@@ -0,0 +1,171 @@
+
+  The Called Methods Checker does not have a way of expressing that a
+  method must never be called.  You can do unsound bug-finding for such a
+  property by using the `!` operator.  The Called Methods Checker will
+  detect if the method was always called, but will silently approve the code
+  if the method is called on some but not all paths.
+
+\item[\refqualclass{checker/CalledMethods/qual}{EnsuresCalledMethods}]
+  This declaration annotation specifies a post-condition on a method, indicating the methods it
+  guarantees to be called on some input expression.
+
+  For example, this specification:
+
+  \begin{Verbatim}
+    @EnsuresCalledMethods(value = "#1", methods = {"x","y"})
+    void m(Param p) { ... }
+  \end{Verbatim}
+
+  guarantees that \<p.x()> and \<p.y()> will always be called before \<m> returns.
+  The body of \<m> must satisfy that property, and clients of \<m> can depend on the property.
+
+\item[\refqualclass{common/returnsreceiver/qual}{This}]
+  \<@This> may only be written on a method return type, and means that the method returns its receiver.
+  This is helpful when type-checking fluent APIs. This annotation is defined by the
+  Returns Receiver Checker (\chapterpageref{returns-receiver-checker}), but is particularly useful
+  for the Called Methods Checker because many builders are fluent APIs.
+
+\end{description}
+
+\sectionAndLabel{Default handling for Lombok and AutoValue}{called-methods-framework-details}
+
+This section gives detailed rules on how the Called Methods Checker infers types for the Lombok
+and AutoValue frameworks. Most readers can skip these details.
+
+The Called Methods Checker automatically assumes default annotations for code that uses builders generated
+by Lombok and AutoValue. There are three places annotations are usually assumed:
+\begin{itemize}
+\item An \<@CalledMethods> annotation is placed on the receiver of the \<build()> method, capturing the
+setter methods that must be invoked on the builder before calling \<build()>. For Lombok,
+this annotation's argument is the set of \<@lombok.NonNull> fields that do not have default values.
+For AutoValue, it is the set of fields that are not \<@Nullable>, \<Optional>, or a Guava Immutable
+Collection.
+\item If the object has a \<toBuilder()> method (for example, if the \<toBuilder = true> option is
+passed to Lombok's \<@Builder> annotation), then the return type of that method is annotated with
+the same \<@CalledMethods> annotation as the receiver of \<build()>, using the same rules as above.
+\item An \<@This> annotation is placed on the return type of each setter in the builder's implementation.
+\end{itemize}
+
+You can disable the framework supports by specifying them in a comma-separated list to the
+command-line flag \<disableFrameworkSupports>.  For example, to disable both Lombok and AutoValue supports,
+use \<-AdisableFrameworkSupports=AutoValue,Lombok>.
```

Comment:
Should we encourage the user to write `-ACalledMethodsChecker_disableFrameworkSupports` instead?

Reply:
Probably. I'll change it - nice catch.